packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,74 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback, useMemo } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+import { get } from "lodash";
+import useStudioStore from "../store/useStudioStore";
+import useRawSiteSettings from "../hooks/useRawSiteSettings";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const expressionSources = useExpressionSources();
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
+      onChange(val);
+    },
+    [onChange]
+  );
+
+  const appendField = useCallback(
+    (fieldId: string) => {
+      const newPropValue = get(
+        { document: expressionSources["document"] },
+        fieldId
+      );
+      onChange(newPropValue);
+    },
+    [onChange, expressionSources]
+  );
+
+  const fieldPickerNumberFilter = useCallback(
```

Comment:
This callback doesn't take in any dependencies so we can put it outside the body of the component like useExpressionSources. We can also update the corresponding one in StringPropInput